pyiron_gui/project/project_browser.py

```diff
@@ -140,40 +495,34 @@ def __getattr__(self, item):
     def clear_output(self, *args, **kwargs):
         clear_button = kwargs.pop('clear_button', False)
         if clear_button:
-            self.buttons = widgets.HBox()
+            self.header = widgets.HBox()
         self.output.clear_output(*args, **kwargs)
         self.refresh()
 
-    def _update_buttons(self, obj):
-
-        def click_button(b):
-            self.output.clear_output()
-            self.display(obj=obj)
-
-        if isinstance(obj, PyironWrapper) and obj.has_self_representation:
-            button = widgets.Button(description="Re-plot " + obj.name)
-            button.on_click(click_button)
-            self.buttons.children = tuple([button])
-
-        self.refresh()
-
     def display(self, obj, default_output=None):
-        self._update_buttons(obj)
+        if isinstance(obj, BaseWrapper):
+            self.display(obj.gui, default_output=default_output)
+        elif isinstance(obj, np.ndarray):
+            self.display(NumpyWidget(obj), default_output=default_output)
+        else:
+            self._display_obj = obj
+            self._display(default_output)
+
+    def _display(self, default_output):
         with self.output:
-            if obj is None and default_output is None:
+            if self._display_obj is None and default_output is None:
                 raise TypeError("Given 'obj' is of 'NoneType'.")
-            elif obj is None:
+            elif self._display_obj is None:
                 print(default_output)
-            elif isinstance(obj, PyironWrapper):
-                plt.ioff()
-                to_display = obj.self_representation()
-                if to_display is not None:
-                    display(obj.self_representation())
+            elif isinstance(self._display_obj, ObjectWidget):
+                display(self._display_obj.gui)
```

Comment:
All `ObjectWidget`s come into existence via the `BaseWrapper` classes (+numpy), right?  When is this triggered then?

Reply:
Actually, all the time. If the `DisplayOutputGUI` is asked to display something derived from a `BaseWrapper` it calls itself with the `gui` property of the `BaseWrapper` which is an instance of an `ObjectWidget`.